### PR TITLE
breaking: Bump Alpine to 3.15 and Java Runtime to 17.x

### LINKED ISSUE #238: Remove use of Java 8
This is to spun conversation...

We are using Java8 for he image and we are already passed 2 LTS from that, v11 (Sept 2018) and v17 (Sept 2021).

I run some tests and:
* Java11 is fine, only upgrading the alpine package to `openjdk11-jre` works fine.
* Java17 is only available in latest Alpine v3.15.0. This raised concerns about https://github.com/asciidoctor/docker-asciidoctor/pull/216, but if I understand correctly this is fixed with Docker engine 20.10.x and that has been out for more than a year (2020-12-08, https://docs.docker.com/engine/release-notes/#20100).
As a extra, upgrading alpine also fixes CVEs (se table below).

My opinion is to upgrade to latests Java LTS and Alpine, unless there's some support reason to still use older Alpine. wdyt?

```
asciidoctor:latest (alpine 3.13.7)
==================================
Total: 14 (UNKNOWN: 0, LOW: 0, MEDIUM: 6, HIGH: 5, CRITICAL: 3)

+----------+------------------+----------+-------------------+---------------+---------------------------------------+
| LIBRARY  | VULNERABILITY ID | SEVERITY | INSTALLED VERSION | FIXED VERSION |                 TITLE                 |
+----------+------------------+----------+-------------------+---------------+---------------------------------------+
| expat    | CVE-2022-22822   | CRITICAL | 2.2.10-r1         | 2.2.10-r2     | expat: Integer overflow in            |
|          |                  |          |                   |               | addBinding in xmlparse.c              |
|          |                  |          |                   |               | -->avd.aquasec.com/nvd/cve-2022-22822 |
+          +------------------+          +                   +               +---------------------------------------+
|          | CVE-2022-22823   |          |                   |               | expat: Integer overflow in            |
|          |                  |          |                   |               | build_model in xmlparse.c             |
|          |                  |          |                   |               | -->avd.aquasec.com/nvd/cve-2022-22823 |
+          +------------------+          +                   +               +---------------------------------------+
|          | CVE-2022-22824   |          |                   |               | expat: Integer overflow in            |
|          |                  |          |                   |               | defineAttribute in xmlparse.c         |
|          |                  |          |                   |               | -->avd.aquasec.com/nvd/cve-2022-22824 |
+          +------------------+----------+                   +               +---------------------------------------+
|          | CVE-2021-45960   | HIGH     |                   |               | expat: Large number of prefixed XML   |
|          |                  |          |                   |               | attributes on a single tag can...     |
|          |                  |          |                   |               | -->avd.aquasec.com/nvd/cve-2021-45960 |
+          +------------------+          +                   +               +---------------------------------------+
|          | CVE-2021-46143   |          |                   |               | expat: Integer overflow               |
|          |                  |          |                   |               | in doProlog in xmlparse.c             |
|          |                  |          |                   |               | -->avd.aquasec.com/nvd/cve-2021-46143 |
+          +------------------+          +                   +               +---------------------------------------+
|          | CVE-2022-22825   |          |                   |               | expat: Integer overflow               |
|          |                  |          |                   |               | in lookup in xmlparse.c               |
|          |                  |          |                   |               | -->avd.aquasec.com/nvd/cve-2022-22825 |
+          +------------------+          +                   +               +---------------------------------------+
|          | CVE-2022-22826   |          |                   |               | expat: Integer overflow in            |
|          |                  |          |                   |               | nextScaffoldPart in xmlparse.c        |
|          |                  |          |                   |               | -->avd.aquasec.com/nvd/cve-2022-22826 |
+          +------------------+          +                   +               +---------------------------------------+
|          | CVE-2022-22827   |          |                   |               | expat: Integer overflow               |
|          |                  |          |                   |               | in storeAtts in xmlparse.c            |
|          |                  |          |                   |               | -->avd.aquasec.com/nvd/cve-2022-22827 |
+----------+------------------+----------+-------------------+---------------+---------------------------------------+
| libblkid | CVE-2021-3995    | MEDIUM   | 2.36.1-r1         | 2.37.3-r0     | util-linux: Unauthorized unmount      |
|          |                  |          |                   |               | of FUSE filesystems belonging         |
|          |                  |          |                   |               | to users with similar uid...          |
|          |                  |          |                   |               | -->avd.aquasec.com/nvd/cve-2021-3995  |
+          +------------------+          +                   +               +---------------------------------------+
|          | CVE-2021-3996    |          |                   |               | util-linux: Unauthorized unmount      |
|          |                  |          |                   |               | of filesystems in libmount            |
|          |                  |          |                   |               | -->avd.aquasec.com/nvd/cve-2021-3996  |
+----------+------------------+          +                   +               +---------------------------------------+
| libmount | CVE-2021-3995    |          |                   |               | util-linux: Unauthorized unmount      |
|          |                  |          |                   |               | of FUSE filesystems belonging         |
|          |                  |          |                   |               | to users with similar uid...          |
|          |                  |          |                   |               | -->avd.aquasec.com/nvd/cve-2021-3995  |
+          +------------------+          +                   +               +---------------------------------------+
|          | CVE-2021-3996    |          |                   |               | util-linux: Unauthorized unmount      |
|          |                  |          |                   |               | of filesystems in libmount            |
|          |                  |          |                   |               | -->avd.aquasec.com/nvd/cve-2021-3996  |
+----------+------------------+          +                   +               +---------------------------------------+
| libuuid  | CVE-2021-3995    |          |                   |               | util-linux: Unauthorized unmount      |
|          |                  |          |                   |               | of FUSE filesystems belonging         |
|          |                  |          |                   |               | to users with similar uid...          |
|          |                  |          |                   |               | -->avd.aquasec.com/nvd/cve-2021-3995  |
+          +------------------+          +                   +               +---------------------------------------+
|          | CVE-2021-3996    |          |                   |               | util-linux: Unauthorized unmount      |
|          |                  |          |                   |               | of filesystems in libmount            |
|          |                  |          |                   |               | -->avd.aquasec.com/nvd/cve-2021-3996  |
+----------+------------------+----------+-------------------+---------------+---------------------------------------+

```

### PR DESCRIPTION
**Pull request by abelsromero**
This change also required bumping Alpine since 'openjdk17' package
is only available in latest alpine (https://pkgs.alpinelinux.org/packages?name=openjdk17&branch=v3.15).

closes #238

### CODE REVIEW

**Comment by abelsromero** [comment]
I understand the MD readme is updated only on release, is that so?

File: updatecli/updatecli.d/alpine-linux.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ sources:
       versionFilter:
         kind: regex
```
Comment (dduportal) [inline updatecli/updatecli.d/alpine-linux.yml:26]:
```suggestion
        kind: semver
```

Comment (dduportal) [reply updatecli/updatecli.d/alpine-linux.yml:26]: - As per: https://www.updatecli.io/docs/plugins/github_release/#_semver , it avoids specifying a regex pattern and keep the expected version format for Alpine while always trying to update to latest

File: updatecli/updatecli.d/alpine-linux.yml [hunk 2]
```diff
@@ -25,7 +25,7 @@ sources:
...
         # Stay on Alpine 3.13 mainline because of https://github.com/asciidoctor/docker-asciidoctor/pull/216
```
Comment (dduportal) [inline updatecli/updatecli.d/alpine-linux.yml:27]:
```suggestion
```
Thanks to your work above :)

File: updatecli/updatecli.d/alpine-linux.yml [hunk 3]
```diff
@@ -25,7 +25,7 @@ sources:
...
-        pattern: '3.13.(\d*)'
+        pattern: '3.15.(\d*)'
```
Comment (dduportal) [inline updatecli/updatecli.d/alpine-linux.yml:28]:
```suggestion
```

If you switch to "kind: semver" above

**Review by dduportal — COMMENTED** [review]
Great work, thanks a lot!

I got a few feedbacks (non blocking though).

I'll also rename the PR title since the changelog is auot-generated from it, to underline that alpine is also bumped and that it's a breaking change.

**Comment by dduportal** [comment]
> I understand the MD readme is updated only on release, is that so?

That is correct. It's only used for DockerHub description.

**Comment by abelsromero** [comment]
Thanks for the feefback, corrections applied

**Pull request merged**